### PR TITLE
Attribute specific filter

### PR DESCRIPTION
**Pull request by Denommus**
## Description

Adds an attribute specific filter

## Related Issue

Refs liftedinit/many-rs#166

## Testing

Added a unit test that checks if the filter works

## Checklist:

- [X] I have read and followed the CONTRIBUTING guidelines for this project.
- [X] I have added or updated tests and they pass.
- [X] I have added or updated documentation and it is accurate.
- [X] I have noted any breaking changes in this module or downstream modules.

### CODE REVIEW

File: src/many-kvstore/src/module/event.rs [hunk 1]
```diff
@@ -93,3 +99,106 @@ fn filter_date<'a>(
...
+                    }) => state.contains(&MultisigTransactionState::Withdrawn),
+                    Ok(EventLog {
+                        content: EventInfo::AccountMultisigExpired { .. },
+                        ..
+                    }) => state.contains(&MultisigTransactionState::Expired),
+                    _ => false,
+                }))
+            }
+        }
+    }
```
Comment (fmorency) [inline src/many-kvstore/src/module/event.rs:142]: FYI, for now, at least, Multisig* has nothing to do with kvstore. 

I don't think we need to support this impl at the moment.


However, you definitely need to add it to `many-ledger`, which is missing.

File: src/many-kvstore/src/module/event.rs [hunk 2]
```diff
@@ -93,3 +99,106 @@ fn filter_date<'a>(
...
+    it
+}
+
+#[cfg(test)]
```
Comment (fmorency) [inline src/many-kvstore/src/module/event.rs:146]: Move that test to `tests/events.rs` and follow the naming convention there.

File: src/many-kvstore/src/module/event.rs [hunk 3]
```diff
@@ -93,3 +99,106 @@ fn filter_date<'a>(
...
+                    Ok(EventLog {
+                        content: EventInfo::AccountMultisigWithdraw { .. },
+                        ..
+                    }) => state.contains(&MultisigTransactionState::Withdrawn),
+                    Ok(EventLog {
+                        content: EventInfo::AccountMultisigExpired { .. },
+                        ..
+                    }) => state.contains(&MultisigTransactionState::Expired),
+                    _ => false,
+                }))
```
Comment (fmorency) [inline src/many-kvstore/src/module/event.rs:139]: Is there a way to check that the internal state is one of the `MultisigTransactionState::*` without having to list them explicitly?

**Review by fmorency — CHANGES_REQUESTED** [review]
Thanks!

File: src/many-kvstore/src/module/event.rs (hunk 3 above)
Comment (Denommus) [reply src/many-kvstore/src/module/event.rs]: I implemented this with @hansl himself, I don't think there's a better way of doing so.

**Review by fmorency — APPROVED** [review]
Yeah. I'll approve it as-is.

Add a TODO, so we don't forget to test the other Txs status.

**Pull request merged**